Add blog post: Building the Neoway's ML Platform with a Team-First Approach and Product Thinking (#32)
NeowayLabs/neowaylabs.github.io: _posts/2022-01-19-neoways-ml-platform.md

    - [User Interviews](#user-interviews)
    - [Proto-Personas](#proto-personas)
    - [User Story Map](#user-story-map)
    - [Prioritization](#prioritization)
    - [Agile Execution](#agile-execution)
- [Key Takeaways](#key-takeaways)

----

[Neoway](http://neoway.com.br) is a Brazilian company that generates intelligence for decision-making and productivity for the processes of marketing, compliance, fraud prevention, legal analysis, and credit management. Since its foundation, the company has invested in a data platform for collection, ingestion, treatment, and presentation that can be used to create products and services for different use cases. 

In the last five years, data science and machine learning have emerged to bring more intelligence in solving customer problems and currently play a key role in the solutions that make up the portfolio. Models, indicators, intelligent journeys are examples of data science-powered applications delivered on our web application or via APIs.

The company's growth brought new challenges to maintain a good communication structure among teams and fast business value delivery to the customers.

# Organizational Structure

> For a long time, the company was organized by functional areas like platform, data, data science, application, product. This kind of organization caused functional silos that led to problems such as lack of systemic thinking, and hard dependency management during software development.

Reviewer: ```suggestion
For a long time, the company was organized by functional areas like platform, data source, data integration, data science, application, product. This kind of organization caused functional silos that led to problems such as lack of systemic thinking, and hard dependency management during software development.
```

Just data team I think is not very clear for english speakers.

For instance, the data science team struggled to get historical data in the proper format for the modeling tasks. The team also had difficulties deploying model predictions integrated with the company's architecture. Both tasks had dependencies with internal teams and the services were not provided properly.

> That organizational structure resulted in inefficiency to deliver business value to our customers. At the end of 2019, the company was redesigned into cross-functional teams focused on business domains capable of delivering end-to-end solutions and accelerating software delivery.

Reviewer: ```suggestion
That organizational structure resulted in inefficiency to deliver business value to our customers. At start of 2020, the company was redesigned into cross-functional teams focused on business domains capable of delivering end-to-end solutions and accelerating software delivery.
```

I remember this differently. If I recall correctly, the team redesign was in the beginning of 2020, no? :thinking: 

@mfrata any opinion here?

## Redesigning our Teams

The organizational structure was changed by choosing business domains to focus on and creating teams to maximize the software delivery to our customers on those domains. The functional teams were changed to cross-functional units of work named **cells**. They are small units of delivery that work collaboratively.

The **stream-aligned** and **platform** are the most fundamental team types in our company. Stream-aligned teams have a continuous flow of work aligned to a business domain. Platform teams provide internal services to reduce the complexity for the stream-aligned teams during software development and delivery.

The image below presents a very summarized view of some cells with which our machine learning platform team interacts the most.

<figure>
  <img src="/images/posts/neoways-ml-platform/cells_overview.png" alt="">
</figure>

The product and service cells are responsible for delivering value directly to the end customers. They need to be empowered to build and deliver customer value as quickly and independently as possible. Those cells are created to better serve the business domain requirements.

> On the other hand, the platform cells have two responsibilities. The first one is to simplify the use of complex technologies that require a high level of specialization, such as Kubernetes, Spark clusters, distributed systems, data lake, streaming platform, IAM, security, data warehouses, search engines. The list goes on.

Reviewer: ```suggestion
On the other hand, the platform cells have two responsibilities. The first one is to simplify the use of complex technologies that require a high level of specialization, such as Kubernetes, Spark clusters, distributed systems, data lake, streaming platform, IAM, security, data warehouses, search engines... The list goes on.
```

I think this punctuation style (ellipsis) makes more sense for a long sequence of list than a period (maybe without end).

A dedicated team with mixed skills in software engineering, infrastructure, and machine learning could stay focused on how to propose solutions to the main problems brought up by data scientists. While data scientists are more focused on business and statistical skills.

## Product Thinking

The first decision was how to approach the challenges above mentioned at that moment. 

A technical team may be tempted to solve the problems only from a technical perspective by considering aspects such as scalability, reliability, and maintainability. Those things are important, but the platform solutions can’t fail at a crucial point: user adoption.

In the blog post [Mind the platform execution gap](https://martinfowler.com/articles/platform-prerequisites.html), the authors have suggested prioritizing developer experience, and it’s one of the aspects to mind the platform execution gap. The reason is simple: a product that no one uses is not a successful product, no matter its technical merits.

Our team decided to approach the platform with product thinking. We decided to focus on the user needs and experience, but deliver solutions with technical excellence. 

> From the beginning, we interviewed the users, tried to understand their journey, and listened to their pain points. It's necessary to see any user perspective on the platform components being developed. An advantage of internal products is that we have closer contact with users through the company’s communication channels. **User empathy is the key!** 

Reviewer: I cannot agree more with this!
